### DISCUSSION TITLE
CustomDevice.json for covers that inappropriately report positions.

### DISCUSSION THREAD

**Discussion by Matt-PMCT** [zwave-js/zwave-js-ui#1127] [Q&A]
I have some exterior shades that we use. They utilize a Somfy RTSI converter to change z-wave into their radio signal. Unfortunately the converter advertises that it can report cover position, but it cannot (it just parrots the last value you send it). I want to get around this by always having Home Assistant see a 50 value which will allow for both opening/closing the cover. 

I thought I could do this by creating a custom device and setting the value template to a number, but Home Assistant is not seeing that 50. Since I'm using the webSocket connection there are no MQTT messages I can look at to troubleshoot, so I'm looking for anyone's advice or suggestions on what I may be doing wrong here. I know the value template works as I tested it in Home Assistant Tempalting and it does spit out a 50. 

```
{
  "71-21505-23122": [
    {
      "type": "cover",
      "object_id": "position",
      "values": ["38-0-currentValue", "38-0-targetValue"],
      "discovery_payload": {
        "command_topic": "38-0-targetValue",
        "position_topic": "38-0-currentValue",
        "set_position_topic": "38-0-targetValue",
        "value_template": "{{ ( 50 ) }}",
        "position_open": 99,
        "position_closed": 0,
        "payload_open": "99",
        "payload_close": "0"
      }
    }
  ]
}
```

**Comment by robertsLando** [comment]
Did you try to just write `"value_template": "50 "`?

**Reply by Matt-PMCT** [reply]
I'm don't think the 50 is the problem. I'm not sure how to update the discovery when not using the MQTT method. With the MQTT I can snoop and see that the discovery is updating as expected, but I don't know how to do that with the WebSocket. It seems as if my customDevice change is not being acted on in the WebSocket connection?

**Reply by robertsLando** [reply]
Mqtt and websocket are two different ways to control your devices with HASS, customDevices is not used by zwave-js server at all. I have no control on what zwave-js server and zwavejs integration do, I only start the zwave-js server passing the driver instance

**Reply by Matt-PMCT** [reply]
Got it, thanks. I didn't realize that custom devices only applied to the MQTT.
